lib/shoulda/matchers/active_model/validate_presence_of_matcher.rb

```diff
@@ -116,12 +138,19 @@ def with_message(message)
           self
         end
 
+        def allow_nil(allow_nil = true)
+          @allow_nil = allow_nil
+          self
+        end
+
         def matches?(subject)
           super(subject)
           @expected_message ||= :blank
 
           if secure_password_being_validated?
             disallows_and_double_checks_value_of!(blank_value, @expected_message)
+          elsif @allow_nil
+            allows_value_of(nil)
```

Comment:
As I understand it, when you `validate_presence_of` a `belongs_to` association it checks to make sure that record exists.  Rails actually queries the DB to make sure that a record exists matching the `belongs_to_id`.